### ISSUE TITLE
Rule Enhancement Proposal: add options to `object-curly-spacing` rule for imports and destructuring assignment

### ISSUE THREAD

**Issue by almirfilho** [eslint/eslint#7361]
**What rule do you want to change?**
Stylistic rule [`object-curly-spacing`](http://eslint.org/docs/rules/object-curly-spacing).

**Does this change cause the rule to produce more or fewer warnings?**
Yes. More or fewer, it depends on the configuration.

**How will the change be implemented? (New option, new default behavior, etc.)?**
With two new options, one for ES6 `import` statement and other for ES6 destructuring assignment:
- `import`: Boolean
- `destructuringAssignment`: Boolean

Today we just have the options `arraysInObjects` and `objectsInObjects`, however, the usage of curly braces may be very different when using ES6's **import statement** and **destructuring assignment**, so it may be desirable to have different settings for each one of those.

**Please provide some example code that this change will affect:**

``` js
import { foo } from 'bar';
let obj = {bar: 'foo'};
const { bar } = obj;
```

**What does the rule currently do for this code?**

``` js
/* eslint object-curly-spacing: ["error", "never"] */

import { foo } from 'bar'; // NOT valid
let obj = {bar: 'foo'};    // valid
const { bar } = obj;       // NOT valid
```

**What will the rule do after it's changed?**

``` js
/* eslint object-curly-spacing: ["error", "never", {"import": true, "destructuringAssignment": true}] */

import { foo } from 'bar'; // valid
let obj = {bar: 'foo'};    // valid
const { bar } = obj;       // valid
```

---

What do you guys think about this?
Thanks!

**Comment by kaicataldo** [comment]
Seems like we should make all three cases individually configurable if we're going to do two

**Comment by almirfilho** [comment]
@kaicataldo I just wanna hear from you guys if it seems a good ideia, then I can work on a PR for that =)

**Comment by kaicataldo** [comment]
Appreciate you being willing to take on the work of implementing this! Let's see what the rest of the team thinks :)

**Comment by vitorbal** [comment]
:+1: from me, but one question: are destructuring assignments and that
syntax of ES6 import technically considered objects? Just wondering if the
rule name wouldn't be misleading for those options, instead of a new rule?

**Comment by kaicataldo** [comment]
I agree that the name is misleading. `curly-spacing` or `curly-bracket-spacing` would be better

**Comment by kaicataldo** [comment]
Closing this as discussion has stalled and it doesn't appear there is enough interest, either from the team or from the community, to continue the discussion at this time. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. Please feel free to open a new issue with a proposal to continue this discussion if this is something you would like to continue advocating for.
